tools/data_dictionary_builder.php

```diff
@@ -94,23 +94,26 @@
 
 $instrumentParameterTypeIDString = implode(', ', $instrumentParameterTypeIDs);
 
-//delete all 'Instrument' entries from parameter_type_category_rel
-$DB->run(
-    "DELETE FROM parameter_type_category_rel
-    WHERE ParameterTypeID in ($instrumentParameterTypeIDString)"
-);
+if (!empty($instrumentParameterTypeIDString)) {
+    //delete all 'Instrument' entries from parameter_type_category_rel
+    $DB->run(
+        "DELETE FROM parameter_type_category_rel
+        WHERE ParameterTypeID in ($instrumentParameterTypeIDString)"
+    );
```

Comment:
I think there should be 3 distinct if !empty checks around each delete statement, since they're using different variables..